ledger/participant-integration-api/src/main/scala/platform/indexer/parallel/ParallelIndexerFactory.scala

```diff
@@ -3,109 +3,92 @@
 
 package com.daml.platform.indexer.parallel
 
+import java.sql.Connection
 import java.util.concurrent.TimeUnit
 
 import akka.NotUsed
 import akka.stream.scaladsl.{Keep, Sink, Source}
 import akka.stream.{KillSwitch, KillSwitches, Materializer, UniqueKillSwitch}
 import com.daml.ledger.participant.state.v1.{Offset, ParticipantId, ReadService, Update}
 import com.daml.ledger.resources.{Resource, ResourceContext, ResourceOwner}
+import com.daml.logging.LoggingContext
 import com.daml.metrics.Metrics
+import com.daml.platform.configuration.ServerRole
 import com.daml.platform.indexer.parallel.AsyncSupport._
 import com.daml.platform.indexer.parallel.PerfSupport._
 import com.daml.platform.indexer.{IndexFeedHandle, Indexer}
+import com.daml.platform.store.DbType
+import com.daml.platform.store.appendonlydao.{DbDispatcher, JdbcLedgerDao}
 import com.daml.platform.store.appendonlydao.events.{CompressionStrategy, LfValueTranslation}
 import com.daml.resources
 
 import scala.concurrent.Future
+import scala.concurrent.duration.FiniteDuration
 import scala.util.control.NonFatal
 
 object ParallelIndexerFactory {
 
   // TODO append-only: migrate code for mutable initialisation
-  def apply(
-      jdbcUrl: String,
+  def apply[DB_BATCH](
+      jdbcUrl: String, // TODO maybe inject the whole dispatcher instead, and let a higher level factory create that alongside with StorageBackend
+      storageBackend: StorageBackend[DB_BATCH],
       participantId: ParticipantId,
       translation: LfValueTranslation,
       compressionStrategy: CompressionStrategy,
       mat: Materializer,
       inputMappingParallelism: Int,
+      batchingParallelism: Int,
       ingestionParallelism: Int,
       submissionBatchSize: Long,
       tailingRateLimitPerSecond: Int,
       batchWithinMillis: Long,
       metrics: Metrics,
-  ): ResourceOwner[Indexer] = {
+  )(implicit loggingContext: LoggingContext): ResourceOwner[Indexer] = {
     for {
       inputMapperExecutor <- asyncPool(
         inputMappingParallelism,
         Some(metrics.daml.parallelIndexer.inputMapping.executor -> metrics.registry),
       )
-      postgresDaoPool <- asyncResourcePool(
-        () => JDBCPostgresDAO(jdbcUrl),
-        size = ingestionParallelism + 1,
-        Some(metrics.daml.parallelIndexer.ingestion.executor -> metrics.registry),
+      batcherExecutor <- asyncPool(
+        batchingParallelism,
+        Some(metrics.daml.parallelIndexer.batching.executor -> metrics.registry),
+      )
+      dbDispatcher <- DbDispatcher.owner( // TODO do we need to wire health status here somehow?
+        serverRole = ServerRole.Indexer,
+        jdbcUrl = jdbcUrl,
+        connectionPoolSize = ingestionParallelism + 1, // + 1 for the tailing ledger_end updates
+        connectionTimeout = FiniteDuration(
+          250,
+          "millis",
+        ), // TODO verify why would we need here a timeout, in this case (amount of parallelism aligned) it should not happen by design
```

Comment:
Canton fork uses this instead:
```
30.seconds, // overridden in canton back to hikari default as the upstream 250ms setting is too low
```

Reply:
Which might be relevant for a high load scenario on the ReadDao side, but here currently I do not see any reason to have this not 0: the pool is exactly the same size how many users it has at the same time. (AFAIK this config boils down to Hikari pool config which defines timeout for getting the connection for a client)
I would keep the TODO, let's start digging into Hikari internals in epic #3